### PR TITLE
metrics separation

### PR DESCRIPTION
**Pull request by drahnr**
Separate the metrics definitions in some very old subsystem implementations, that haven't been touch in a while.

Mostly an extraction of #5314 to get that timely merged and avoid merge-pain.

### CODE REVIEW

**Review by ordian — APPROVED** [review]

**Comment by eskimor** [comment]
Gladly you left collator-protocol untouched. (I am working on a PR that did the same thing for the collator protocol).

**Review by eskimor — APPROVED** [review]

**Comment by drahnr** [comment]
> 

I touched it _too much_ and could not extract the required changeset cleanly :cold_sweat: merge conflict upcoming..

File: node/collation-generation/src/metrics.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
```
Comment (slumber) [inline node/collation-generation/src/metrics.rs:1]: 2022

_I just had to_

**Review by slumber — APPROVED** [review]

File: node/collation-generation/src/metrics.rs (hunk 1 above)
Comment (drahnr) [reply node/collation-generation/src/metrics.rs:1]: The source code was just copied, hence left as is - copying, in parts or full, does not alter the copyright.

**Comment by drahnr** [comment]
bot merge

**Pull request merged**